src/org/traccar/model/Calendar.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright 2016 Anton Tananaev (anton@traccar.org)
+ * Copyright 2016 Andrey Kunitsyn (andrey@traccar.org)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.traccar.model;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.sql.SQLException;
+import java.util.Collection;
+import java.util.Date;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+
+import net.fortuna.ical4j.data.CalendarBuilder;
+import net.fortuna.ical4j.data.ParserException;
+import net.fortuna.ical4j.filter.Filter;
+import net.fortuna.ical4j.filter.PeriodRule;
+import net.fortuna.ical4j.filter.Rule;
+import net.fortuna.ical4j.model.Component;
+import net.fortuna.ical4j.model.DateTime;
+import net.fortuna.ical4j.model.Dur;
+import net.fortuna.ical4j.model.Period;
+import net.fortuna.ical4j.model.component.CalendarComponent;
+import net.fortuna.ical4j.validate.ValidationException;
+
+public class Calendar extends Extensible {
+
+    private String name;
+
+    public String getName() {
+        return name;
+    }
+
+    public void setName(String name) {
+        this.name = name;
+    }
+
+    private byte[] calendarData;
+
+    public byte[] getCalendarData() throws ValidationException, IOException {
+        return calendarData.clone();
+    }
+
+    public void setCalendarData(byte[] calendarData) throws IOException, ParserException, SQLException {
+        setCalendar(calendarData);
+    }
+
+    private net.fortuna.ical4j.model.Calendar calendar;
+
+    @JsonIgnore
+    public net.fortuna.ical4j.model.Calendar getCalendar() {
+        return calendar;
+    }
+
+    public void setCalendar(byte[] calendarData) throws IOException, ParserException, SQLException {
+        CalendarBuilder builder = new CalendarBuilder();
+        calendar = builder.build(new ByteArrayInputStream(calendarData));
+        this.calendarData = calendarData.clone();
+    }
+
+    public boolean checkMoment(Date date) {
+        if (calendar != null) {
+            Period period = new Period(new DateTime(date), new Dur(0, 0, 0, 1));
```

Comment:
Yes, period of one second length.
rfc-2445 has paradigm of requests for events and free/busy states. Request has to have startTime and endTime, or startTime and duration. Filtering is some kind of request.

Do not think any implementation of rfc 2445 has functionality to check moment of time, not period.

Theoretically, we could reinvent it, iterate all vevents and check if moment in between start and end.
But firstly we have to request such vevents with some kind of request, that contain at least start date. It will be double work.